### PR TITLE
[ByteTrack] - add a method to reset the state of the tracker

### PR DESCRIPTION
**Pull request by AdonaiVera**
## Description

This Pull Request introduces a `reset` method to the ByteTrack tracking system, enabling efficient processing of multiple video streams sequentially within the same execution environment. The enhancement ensures that internal tracking states such as `tracked_tracks`, `lost_tracks`, `removed_tracks`, and the global `BaseTrack._count` are reset between videos, addressing inaccuracies caused by state persistence.

The addition is particularly useful for applications requiring sequential video analysis without the overhead of restarting the Python environment, streamlining workflows in surveillance, traffic monitoring, and activity recognition.

**Issue Addressed:** #789 

## Type of Change

- [x] New feature (non-breaking change which adds functionality)

## Testing

The implementation was tested using a new test case `ultralytics_example_multiple_videos`, processing multiple video streams to verify the tracker's state resets correctly. Additionally, a Google Colab demo illustrates the reset functionality in action.

- **Google Colab Demo:**[ [View Demo]](https://colab.research.google.com/drive/1i8e9WURNDm8FzbHTi3ljJtuWycugKmqw?usp=sharing)
- **Test Video Results:**

https://github.com/roboflow/supervision/assets/45982251/52a13c59-63d1-48b3-97f5-4194996201e3

https://github.com/roboflow/supervision/assets/45982251/a71d6ae9-c8c9-4321-9044-e7ea4d00c79d

## Deployment Considerations

No specific considerations. The `reset` method is backward compatible and should seamlessly integrate into existing deployments.

## Documentation Updates

- **Reset Method:** Documented the usage and impact of the `reset` method within the ByteTrack class.
- **Testing Procedure:** Updated the `ultralytics_example_multiple_videos` test case documentation to reflect the sequential video processing testing approach.

### CODE REVIEW

**Comment by SkalskiP** [comment]
Hi @AdonaiVera 👋🏻 Please drop `ultralytics_example_multiple_videos` script. 🙏🏻

**Comment by AdonaiVera** [comment]
I just removed it @SkalskiP

File: supervision/tracker/byte_tracker/core.py [hunk 1]
```diff
@@ -261,6 +260,13 @@ def callback(frame: np.ndarray, index: int) -> np.ndarray:
 
         return detections
 
+    def reset(self):
```
Comment (SkalskiP) [inline supervision/tracker/byte_tracker/core.py:263]: Can we document this method? For the method to be displayed in docs it needs to be public and have docstring.

**Review by SkalskiP — CHANGES_REQUESTED** [review]

**Comment by SkalskiP** [comment]
I tested with your colab. Looks like it works. As I mentioned in the comment, we need a docstring for your newly added method. I also see that pre-commit is [complaining](https://results.pre-commit.ci/run/github/571613891/1706736422.ALwQ5a5NTOyy0c3Zl_4tNw) about file formatting.

File: supervision/tracker/byte_tracker/core.py (hunk 1 above)
Comment (AdonaiVera) [reply supervision/tracker/byte_tracker/core.py:263]: Hi @SkalskiP Thank you for the suggestions. I have already implemented the changes. Please let me know if there is anything else that needs to be updated. 🥷 🚀

**Comment by SkalskiP** [comment]
Thank you @AdonaiVera 🙏🏻 Awesome. I tested the solution. Everything works. Let's merge.

**Review by SkalskiP — APPROVED** [review]

**Pull request merged**